utils/logger.go

```diff
@@ -0,0 +1,109 @@
+// this is a new logger interface for mattermost
+
+package utils
```

Comment:
The context stuff can probably stay in utils for the time being, but I agree that the logger itself could have its own package.